### PR TITLE
Override alias template to preserve anchor fragment

### PR DESCRIPTION
**Pull request by jolheiser**
This PR fixes an annoyance where docs aliases aren't preserving their anchor fragments.

The refactor included aliases to keep old links from dying, but currently they redirect without their anchor, which was used _often_ to jump to sections.

This overrides the alias template with an alternative that preserves the anchor fragment.

To note, this is just a copy of the [embedded template](https://github.com/gohugoio/hugo/blob/5c7b79cf7f00aa2651dd5f0364ee575af6715a31/tpl/tplimpl/embedded/templates/alias.html), but defaults to a JS redirect that preserves the anchor, and uses the meta tag as a fallback for noscript users.

I've also preserved query params, but I can remove that if we don't think it's necessary.

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

File: docs/layouts/alias.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html{{ with site.LanguageCode | default site.Language.Lang }} lang="{{ . }}"{{ end }}>
+  <head>
+    <title>{{ .Permalink }}</title>
+    <link rel="canonical" href="{{ .Permalink }}">
+    <meta name="robots" content="noindex">
+    <meta charset="utf-8">
```
Comment (silverwind) [inline docs/layouts/alias.html]:
```suggestion
    <meta charset="utf-8">
    <title>{{ .Permalink }}</title>
    <link rel="canonical" href="{{ .Permalink }}">
    <meta name="robots" content="noindex">
```

Probably mostly superficial, but I prefer `charset` first so the browser can immediately switch into "utf-8" mode.

Comment (jolheiser) [reply docs/layouts/alias.html]: The browser barely even loads this page before redirecting, does the charset do much?

Comment (silverwind) [reply docs/layouts/alias.html]: Likely not, it's just a habit that I follow so stuff like `<title>` does not parse weirdly when the browser was not yet told the charset to use. Likely makes zero difference.

Comment (silverwind) [reply docs/layouts/alias.html]: As per [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#charset):

> meta elements which declare a character encoding must be located entirely within the first 1024 bytes of the document.

So there is a theoretical possibilty it moves outside this boundary when the title is 1kB long 😆

File: docs/layouts/alias.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <noscript><meta http-equiv="refresh" content="0; url={{ .Permalink }}"></noscript>
```
Comment (delvh) [inline docs/layouts/alias.html:8]: Wait, such a use of `<noscript>` is possible?

**Review by delvh — APPROVED** [review]
I think it would be best if this was fixed upstream, but while that's not done I'm fine with that workaround.

**Comment by jolheiser** [comment]
> I think it would be best if this was fixed upstream, but while that's not done I'm fine with that workaround.

I can take a look and propose it upstream as well. 👍 
But this is currently happening, so would be good to merge for now either way.

**Pull request merged**